src/controllers/csi/provisioner/controller.go

```diff
@@ -76,6 +83,34 @@ func (provisioner *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error
 }
 
 func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	fs := afero.NewOsFs()
+	if provisioner.parallelDownloadsEnabled() {
+		if provisioner.parallelDownloadsLimitReached() {
+			log.Info("max parallel downloads reached, requeued")
+			return reconcile.Result{RequeueAfter: shortRequeueDuration}, nil
+		}
+		atomic.AddInt64(&provisioner.currentParallelDownloads, 1)
+		log.Info("staring parallel download")
+		go func() {
```

Comment:
What if reconciliation failed?

Reply:
we will be sad, but we don't care, the logs will be still there

(this is not meant to be a permanent long-term solution and won't be documented so not everyone will use it)

Reply:
It changes the logic of `OneAgentProvisioner`. [Currently](https://github.com/Dynatrace/dynatrace-operator/pull/656/files/d2fa7a67887e6ec71a4db5032baccfee711ccff1#diff-f2b8b611648fec932e0790d9a1ae99ad30a8ca183edce27761bbe1271eb71b89L138) it returns an `error` so that request is re-queued to be processed again.
I'd suggest to preserve this logic by adding an atomic bool which indicates if error occurred in any of goroutines.
If that bool==true, return `reconcile.Result{}, errors.New("parallel download failed")`

Reply:
we would need to block(which we can't) OR log the error in another reconcile, but because we are running stuff in parallel we might override that error, or need to store multiple of these bools.

this shouldn't be over-engineered